inst/stan/epinowcast.stan

```diff
@@ -163,48 +175,92 @@ model {
       rd_eff_sd ~ zero_truncated_normal(rd_eff_sd_p[1], rd_eff_sd_p[2]);
     } 
   }
+  // share of cases with eventually known reference date
+  alpha_sd ~ normal(alpha_sd_p[1], alpha_sd_p[2]) T[0, ];
+  for (k in 1:g){
+    alpha_start[k] ~ normal(alpha_start_p[1], alpha_start_p[2]);
+    alpha_epsilon[k] ~ normal(0, alpha_sd);
+  }
   // reporting overdispersion (1/sqrt)
   sqrt_phi ~ normal(sqrt_phi_p[1], sqrt_phi_p[2]) T[0,];
   }
   // log density: observed vs model
   if (likelihood) {
     profile("model_likelihood") {
-    target += reduce_sum(obs_lupmf, st, 1, flat_obs, sl, csl, imp_obs, sg, st,
-                         rdlurd, srdlh, ref_lh, dpmfs, ref_p, phi);
+    target += obs_lupmf(flat_obs | obs_miss, dmax, sl, csl, g, imp_obs, sg, st,
+                         rdlurd, srdlh, ref_lh, dpmfs, ref_p, alpha, phi);
     }
   }
 }
 
 generated quantities {
```

Comment:
The generated quantities are a bit of a mess / rather inefficient still and I clearly haven't helped things by adding the missing reference date stuff...

Reply:
🙈